src/sentry/models/outbox.py

```diff
@@ -242,6 +489,11 @@ def next_schedule(self, now: datetime.datetime) -> datetime.datetime:
         return now + min((self.last_delay() * 2), datetime.timedelta(hours=1))
 
     def save(self, **kwds: Any) -> None:  # type: ignore[override]
+        if OutboxCategory(self.category) not in _outbox_categories_for_scope[int(self.shard_scope)]:
```

Comment:
And this ensures that every outbox saved follows the heirarchy of scope -> categories.